src/ws.cpp

```diff
@@ -348,27 +348,30 @@ bool Instance::initialize(EGLDisplay eglDisplay)
         return false;
     }
 
+    if (wl_display_init_shm(m_display))
+        return false;
+
     const char* extensions = eglQueryString(eglDisplay, EGL_EXTENSIONS);
     if (isEGLExtensionSupported(extensions, "EGL_WL_bind_wayland_display")) {
         s_eglBindWaylandDisplayWL = reinterpret_cast<PFNEGLBINDWAYLANDDISPLAYWL>(eglGetProcAddress("eglBindWaylandDisplayWL"));
         assert(s_eglBindWaylandDisplayWL);
         s_eglQueryWaylandBufferWL = reinterpret_cast<PFNEGLQUERYWAYLANDBUFFERWL>(eglGetProcAddress("eglQueryWaylandBufferWL"));
         assert(s_eglQueryWaylandBufferWL);
     }
-    if (!s_eglBindWaylandDisplayWL || !s_eglQueryWaylandBufferWL)
-        return false;
 
     if (isEGLExtensionSupported(extensions, "EGL_KHR_image_base")) {
         s_eglCreateImageKHR = reinterpret_cast<PFNEGLCREATEIMAGEKHRPROC>(eglGetProcAddress("eglCreateImageKHR"));
         assert(s_eglCreateImageKHR);
         s_eglDestroyImageKHR = reinterpret_cast<PFNEGLDESTROYIMAGEKHRPROC>(eglGetProcAddress("eglDestroyImageKHR"));
         assert(s_eglDestroyImageKHR);
     }
-    if (!s_eglCreateImageKHR || !s_eglDestroyImageKHR)
-        return false;
 
-    if (!s_eglBindWaylandDisplayWL(eglDisplay, m_display))
-        return false;
+    if (s_eglBindWaylandDisplayWL) {
```

Comment:
Does this work without eglQueryWaylandBuffer? we were checking both before

Reply:
If one is available, the other should be too. But I'll readd the check.